core/src/scene/stops.cpp

```diff
@@ -11,33 +11,48 @@
 
 namespace Tangram {
 
-Stops::Stops(const YAML::Node& _node, bool _isColor) {
+auto Stops::Color(const YAML::Node& _node) -> Stops {
+    Stops stops;
+    if (!_node.IsSequence()) { return stops; }
 
-    if (!_node.IsSequence()) { return; }
+    for (const auto& frameNode : _node) {
+        if (!frameNode.IsSequence() || frameNode.size() != 2) { continue; }
+        float key = frameNode[0].as<float>();
+
+        // parse color from sequence or string
+        uint32_t color = 0;
+        YAML::Node colorNode = frameNode[1];
+        if (colorNode.IsScalar()) {
+            color = CSSColorParser::parse(colorNode.as<std::string>()).getInt();
+        } else if (colorNode.IsSequence() && colorNode.size() >= 3) {
+            float alpha = colorNode.size() > 3 ? colorNode[3].as<float>() : 1.f;
+            color = alpha * 255.;
+            color = (color << 8) + colorNode[2].as<float>() * 255.;
+            color = (color << 8) + colorNode[1].as<float>() * 255.;
+            color = (color << 8) + colorNode[0].as<float>() * 255.;
+        }
+        stops.frames.emplace_back(key, color);
+    }
+    return stops;
+}
+
+auto Stops::Width(const YAML::Node& _node) -> Stops {
+    Stops stops;
+    if (!_node.IsSequence()) { return stops; }
 
     for (const auto& frameNode : _node) {
         if (!frameNode.IsSequence() || frameNode.size() != 2) { continue; }
         float key = frameNode[0].as<float>();
-        if (_isColor) {
-            // parse color from sequence or string
-            uint32_t color = 0;
-            YAML::Node colorNode = frameNode[1];
-            if (colorNode.IsScalar()) {
-                color = CSSColorParser::parse(colorNode.as<std::string>()).getInt();
-            } else if (colorNode.IsSequence() && colorNode.size() >= 3) {
-                float alpha = colorNode.size() > 3 ? colorNode[3].as<float>() : 1.f;
-                color = alpha * 255.;
-                color = (color << 8) + colorNode[2].as<float>() * 255.;
-                color = (color << 8) + colorNode[1].as<float>() * 255.;
-                color = (color << 8) + colorNode[0].as<float>() * 255.;
-            }
-            frames.emplace_back(key, color);
-        } else {
-            // parse distance from string
-            float value = frameNode[1].as<float>();
-            frames.emplace_back(key, value);
+
+        StyleParam::Width widthValue;
+        widthValue.unit = Unit::meter;
+        size_t start = 0;
+
+        if (StyleParam::parseValueUnitPair(frameNode[1].Scalar(), start, widthValue)){
+            stops.frames.emplace_back(key, widthValue);
```

Comment:
I think we can normalize widths into "pixels" here - since widths are paired with a zoom, the conversion is unambiguous and we don't need to keep their units around.